### PR TITLE
chore(Caching): clean up deprecated cache sweeping

### PR DESCRIPTION
**Pull request by ckohen**
**Please describe the changes this PR makes and why it should be merged:**

The semver:major half of #6825 

**Status and versioning classification:**

- I know how to update typings and have done so, or typings don't need updating
- This PR includes breaking changes (methods removed or renamed, parameters moved or removed)

### CODE REVIEW

File: typings/index.test-d.ts [hunk 1]
```diff
@@ -103,15 +102,6 @@ const client: Client = new Client({
...
     Message: 100,
-    ThreadManager: {
-      maxSize: 1000,
-      keepOverLimit: (x: ThreadChannel) => x.id === '123',
-      sweepInterval: 5000,
-      sweepFilter: LimitedCollection.filterByLifetime({
-        getComparisonTimestamp: (x: ThreadChannel) => x.archiveTimestamp ?? 0,
-        excludeFromSweep: (x: ThreadChannel) => !x.archived,
-      }),
-    },
```
Comment (kyranet) [inline typings/index.test-d.ts:114]: Shouldn't we keep this to validate the object overload without `sweep*` options?

Comment (ckohen) [reply typings/index.test-d.ts:114]: whoops, yeah, meant to add the new example here

File: src/util/Options.js [hunk 2]
```diff
@@ -253,18 +222,6 @@ class Options extends null {
...
-        sweepFilter: require('./Util').archivedThreadSweepFilter(),
-      },
-      GuildChannelManager: {
-        sweepInterval: 3600,
-        sweepFilter: require('./Util').archivedThreadSweepFilter(),
-      },
-      ThreadManager: {
-        sweepInterval: 3600,
-        sweepFilter: require('./Util').archivedThreadSweepFilter(),
-      },
```
Comment (vladfrangu) [inline src/util/Options.js:267]: Just to be sure, are these included already?

**Comment by iCrawl** [comment]
This needs a rebease.

File: src/util/Options.js (hunk 2 above)
Comment (ckohen) [reply src/util/Options.js:274]: What do you mean by that?

Comment (vladfrangu) [reply src/util/Options.js:274]: Do the new sweeper options also autosweep archived threads like the old sweepers?

Comment (ckohen) [reply src/util/Options.js:274]: Yes, its in 1 key now though (nice benefit)

**Review by vladfrangu — APPROVED** [review]

**Review by kyranet — APPROVED** [review]

**Review by iCrawl — APPROVED** [review]

**Pull request merged**